src/Engine/Utils.java

```diff
@@ -198,22 +198,32 @@ public static boolean isPathValid(Unit unit, Path path, GameMap map)
     return canReach;
   }
 
-  /**
-   * Calculate the shortest path for unit to take from its current location to map(x, y), and populate
-   * the path parameter with those waypoints.
-   * If no valid path is found, an empty Path will be returned.
-   */
+  /** Alias for {@link #findShortestPath(Unit, int, int, GameMap, boolean) findShortestPath(Unit, int, int, GameMap, boolean)} **/
+  public static Path findShortestPath(Unit unit, XYCoord destination, GameMap map, boolean theoretical)
+  {
+    return findShortestPath(unit, destination.xCoord, destination.yCoord, map, theoretical);
+  }
+  /** Alias for {@link #findShortestPath(Unit, int, int, GameMap, boolean) findShortestPath(Unit, int, int, GameMap, boolean=false)} **/
   public static Path findShortestPath(Unit unit, XYCoord destination, GameMap map)
   {
-    return findShortestPath(unit, destination.xCoord, destination.yCoord, map);
+    return findShortestPath(unit, destination.xCoord, destination.yCoord, map, false);
+  }
+  /** Alias for {@link #findShortestPath(Unit, int, int, GameMap, boolean) findShortestPath(Unit, int, int, GameMap, boolean=false)} **/
+  public static Path findShortestPath(Unit unit, int x, int y, GameMap map)
+  {
+    return findShortestPath(unit, x, y, map, false);
   }
-
   /**
-   * Calculate the shortest path for unit to take from its current location to map(x, y), and populate
-   * the path parameter with those waypoints.
+   * Calculate and return the shortest path for unit to take from its current location to map(x, y).
+   * The Path will avoid non-allied units unless `theoretical` is true.
    * If no valid path is found, an empty Path will be returned.
+   * @param unit The unit under consideration. The Unit's current location will be the starting point for the path,
+   *             and the unit's move-power will limit the path length.
+   * @param destination The desired endpoint for the Path.
+   * @param map The current GameMap referenced by the Path returned.
+   * @param theoretical If true, ignores other Units and move-power limitations.
    */
-  public static Path findShortestPath(Unit unit, int x, int y, GameMap map)
+  public static Path findShortestPath(Unit unit, int x, int y, GameMap map, boolean theoretical)
```

Comment:
Hah. I took a look at your branch, and the `theoretical` fix looked useful. I would have cherry-picked it over, but it wasn't in its own commit. Boo hoo.

Reply:
I seem to have a habit of committing things along with the fixes that support them.
You will see that a lot if you look at AWBWTester